### PR TITLE
Add Did metadata functionality

### PR DESCRIPTION
**Pull request by drbh**
Allow developers to access the listMyDidsWithMeta and setDidMetadata of the indy sdk for Android

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -309,6 +309,17 @@ const indy = {
     throw new Error(`Unsupported operation! Platform: ${Platform.OS}`)
   },
 
+  async listMyDidsWithMeta(wh) {
+    return IndySdk.listMyDidsWithMeta(wh)
```
Comment (TimoGlastra) [inline src/index.js:316]: It seems like the iOS version is not implemented yet. Could you add the following error to indicate this:

```js
if (Platform.OS === 'ios') {
  throw new Error(`Unsupported operation! Platform: ${Platform.OS}`)
}
```

File: src/index.js [hunk 2]
```diff
@@ -309,6 +309,17 @@ const indy = {
...
+  },
+    
+  async setDidMetadata(wh, did, metadata) {
+    if (Platform.OS === 'ios') {
+      return IndySdk.setDidMetadata(metadata, did, wh)
+    }
```
Comment (TimoGlastra) [inline src/index.js]: Same here, I don't think it is implemented yet

**Review by TimoGlastra — COMMENTED** [review]
Thanks @drbh! Have a small suggestion for the iOS side

File: src/index.js (hunk 2 above)
Comment (drbh) [reply src/index.js]: Just added those, and it looks like they have the same functionality in the ios wrapper - albeit with a different name 

https://github.com/hyperledger/indy-sdk/blob/7f4fbb055478181761347a981544fb823d6f47ba/wrappers/ios/libindy-pod/Indy/Wrapper/IndyDid.mm#L125-L137

File: src/index.js (hunk 1 above)
Comment (drbh) [reply src/index.js:316]: seems like it's implemented 

https://github.com/hyperledger/indy-sdk/blob/7f4fbb055478181761347a981544fb823d6f47ba/wrappers/ios/libindy-pod/Indy/Wrapper/IndyDid.mm#L160-L165

but I'm not 100% sure an don't have a good way to test IOS at the moment. Thanks

Comment (TimoGlastra) [reply src/index.js:316]: I mean in the iOS part of this library. See https://github.com/absaoss/rn-indy-sdk/tree/master/ios/IndySdk.swift

Or am I missing something?

Comment (drbh) [reply src/index.js:316]: whoops you're 100% right. I'll update the code to throw an error. Thanks for the feedback! 🙏

File: src/index.js (hunk 2 above)
Comment (drbh) [reply src/index.js]: ^ same as above. Thanks

Comment (drbh) [reply src/index.js]: resolved

**Comment by drbh** [comment]
@TimoGlastra changes made, please let me know if theres anything else I should update before merging thank! 🙂

**Comment by TimoGlastra** [comment]
Thanks @drbh. The changes look good to me.

I'm not able to approve/merge this PR, so we'll need to wait for @jakubkoci to finish up

**Comment by jakubkoci** [comment]
Thanks, @drbh for your contribution!

**Pull request merged**